### PR TITLE
polok docker task 1

### PR DESCRIPTION
**Pull request by codefresher32**
Fixes #

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
- Building docker image & run container

## What was done
- Created a docker file
- Docker file created in such a way that it is optimized
- For each line in docker file comment added like ased

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: react_client/todo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#  - using node becuase it is the required runtime 
+FROM node:alpine
+
+# Defining a workir so our apllication code is in a specific folder for organizing our container
+WORKDIR /app
+
+# Copying package.json as it includes project defination our our react project and is required to run the project
+# Only copying this file specifily 
+#   because this file is the least likely to change
+COPY package.json .
```
Comment (musfique-m) [inline react_client/todo/Dockerfile:11]: Please also copy package-lock.json also to keep track of the exact version of every package that is installed.
Also use wild card to reduce the number of layers.

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: react_client/todo/Dockerfile (hunk 1 above)
Comment (codefresher32) [reply react_client/todo/Dockerfile]: Added package-lock.json as wildcard....

**Review by musfique-m — CHANGES_REQUESTED** [review]
Please remove the files that are not related to docker from this pr.
**Only** docker related files should be a part of this pr.

**Comment by codefresher32** [comment]
> Please remove the files that are not related to docker from this pr. **Only** docker related files should be a part of this pr.

Vi, I just organized the file structure for conflicts issue for merging in react-master branch. As I am copying all the files in dockerfile, I think I need all the files. Can you tell me what I miss?

**Comment by musfique-m** [comment]
> > Please remove the files that are not related to docker from this pr. **Only** docker related files should be a part of this pr.
> 
> Vi, I just organized the file structure for conflicts issue for merging in react-master branch. As I am copying all the files in dockerfile, I think I need all the files. Can you tell me what I miss?

In the `File Changed` section of this pr you can see that there are 21 files changed. If we proceed to merge this branch with the master, all of this file change will be blamed under this pr. That git blame will be almost useless. It will convey little to no information ( sometimes misinformation ) about why these files were changed. 

In the merge conflict, probably you have introduces some new changes while merging with master branch was being done. It resulted in the increase of the number of file changed. This pr should only consist of files that are related to docker.

**Comment by codefresher32** [comment]
Thank you vi. I understand it now. Reverted the commit. Please review this pr.

**Pull request merged**